lib/middleman/s3_sync.rb

```diff
@@ -58,12 +58,22 @@ def update_bucket_versioning
       end
 
       def connection
-        @connection ||= Fog::Storage::AWS.new({
-          :aws_access_key_id => s3_sync_options.aws_access_key_id,
-          :aws_secret_access_key => s3_sync_options.aws_secret_access_key,
+
+        connection_options = {
           :region => s3_sync_options.region,
           :path_style => s3_sync_options.path_style
-        })
+        }
+
+        if s3_sync_options.aws_access_key_id && s3_sync_options.aws_secret_access_key
```

Comment:
My implementation follow official aws-sdk for Ruby closer: there fallback to IAM used if credentials are not specified. 



Best regards,
Kirill Shirinkin

On Thu, Apr 16, 2015 at 4:57 PM, Frederic Jean <notifications@github.com>
wrote:

>>            :region => s3_sync_options.region,
>>            :path_style => s3_sync_options.path_style
>> -        })
>> +        }
>> +
>> +        if s3_sync_options.aws_access_key_id && s3_sync_options.aws_secret_access_key
> I think that we need to flip the conditions. We can create a `use_iam_profile` option that would configure fog to use the IAM profile and otherwise fall back to the old authentication approach.
> That said, detecting which credentials are available (whether it is via the IAM profile, specified options or through environment variables) is probably the right path to take.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/fredjean/middleman-s3_sync/pull/85/files#r28516685